### PR TITLE
feat(testing): Installs Jest, setup config and  adds cache test case

### PR DESCRIPTION
**Pull request by dimitrianoudi**
It installs Jest, sets config and adds cache test cases.

### CODE REVIEW

File: src/__ tests __/MemoryCache.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+        expect(added).toBe(true);
+        expect(listener).not.toHaveBeenCalled();
+    });
+    test('should notify the listener when a value is set', () => {
+        const listener = jest.fn();
+        cache.subscribe(listener, testCollection, testKey);
+        cache.set(testCollection, testKey, testValue);
+
+        expect(listener).toHaveBeenCalledWith(testKey, testValue, undefined);
```
Comment (kearfy) [inline src/__ tests __/MemoryCache.spec.tsx:38]: It might be interesting to trigger a second update here with a change to the cached value to validate that the third argument passed represents the old cache value instead of `undefined`. (`undefined` is only passed if the cache did not yet exist)

Comment (kearfy) [reply src/__ tests __/MemoryCache.spec.tsx:38]: nevermind, I see you are already testing this. Champ!

File: src/__ tests __/MemoryCache.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    });
+
+    // Edge case tests
+    test('get method should return undefined for a non-existent key', () => {
+        const nonExistentKey = 'nonExistentKey';
+        const value = cache.get(testCollection, nonExistentKey);
+
+        expect(value).toBeUndefined();
+    });
+    test('should allow adding a listener to a non-existent key', () => {
```
[the remark this answers is not in the document]
Comment (dimitrianoudi) [reply src/__ tests __/MemoryCache.spec.tsx:62]: Yeah indeed, well that's what the tests cases are for, hehe.
